### PR TITLE
JNG-4179 min max sum avg first last front back with lambda

### PR DESCRIPTION
**Pull request by gprivi**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4179" title="JNG-4179" target="_blank">StoryJNG-4179</a>  min, max, sum, avg, first, last, front, back with lambda
  </td></table>
  <br />

### CODE REVIEW

**Review by gaborflorian — APPROVED** [review]

File: judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java [hunk 1]
```diff
@@ -620,6 +620,25 @@ private String getJql(final LambdaFunction it) {
         if (it == null) {
             return null;            
         }
+        
+        String functionName = it.getFunctionDeclarationReference().getName();
+
+        if (functionName.equals("first")) {
+            return "heads" + "(" + it.getLambdaArgument().getName() + " | " + getJql(it.getExpression()) + ")";
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java:627]: "heads" + "(" => "heads("

File: judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java [hunk 2]
```diff
@@ -620,6 +620,25 @@ private String getJql(final LambdaFunction it) {
...
+        }
+
+        if (functionName.equals("last")) {
+            return "heads" + "(" + it.getLambdaArgument().getName() + " | " + getJql(it.getExpression()) + " DESC)";
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java:631]: "heads" + "(" => "heads("

File: judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java [hunk 3]
```diff
@@ -620,6 +620,25 @@ private String getJql(final LambdaFunction it) {
...
+        }
+
+        if (functionName.equals("front")) {
+            return "tails" + "(" + it.getLambdaArgument().getName() + " | " + getJql(it.getExpression()) + " DESC)";
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java:635]: "tails" + "(" => "tails("

File: judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java [hunk 4]
```diff
@@ -620,6 +620,25 @@ private String getJql(final LambdaFunction it) {
...
+        }
+
+        if (functionName.equals("back")) {
+            return "tails" + "(" + it.getLambdaArgument().getName() + " | " + getJql(it.getExpression()) + ")";
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java:639]: "tails" + "(" => "tails("

**Review by bencegelei — CHANGES_REQUESTED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**